pkg/icingaredis/own_heartbeat.go

```diff
@@ -0,0 +1,92 @@
+package icingaredis
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/go-redis/redis/v8"
+	"github.com/icinga/icingadb/pkg/com"
+	"github.com/icinga/icingadb/pkg/logging"
+	"github.com/icinga/icingadb/pkg/utils"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"strconv"
+	"time"
+)
+
+const statSecs = 15 * 60
+
+// Stats of sync by kind for self-monitoring.
+var Stats = struct {
+	Config  *com.Stat `json:"config"`
+	State   *com.Stat `json:"state"`
+	History *com.Stat `json:"history"`
+	Overdue *com.Stat `json:"overdue"`
+}{
+	com.NewStat(statSecs),
+	com.NewStat(statSecs),
+	com.NewStat(statSecs),
+	com.NewStat(statSecs),
+}
+
+// haer abstracts HA to prevent package import cycles.
+type haer interface {
```

Comment:
I really had to look twice to understand that name. :sweat_smile: Just calling it `HA` wouldn't do any harm, also exporting it doesn't matter, you could even consider this being nice as it's used as parameter type of an exported function.

Reply:
I already considered intentionally adding a typo: hare 🐰 :P